pkgs/development/libraries/arcan/default.nix

```diff
@@ -0,0 +1,136 @@
+{ stdenv, lib, fetchFromGitHub, cmake, pkg-config, xz, libuvc, libffi, libX11
+, libxcb, libXcomposite, libXfixes, xcbutil, xcbutilwm, libXau, libXdmcp, openal
+, freetype, sqlite, libGL, libdrm, libxkbcommon, mesa, valgrind, luajit, libusb1
+, harfbuzzFull, SDL2
+
+# For generating API documentation
+, ruby
+
+, withVLC ? true, libvlc
+
+#
+, withFFmpeg ? true, ffmpeg-full
+
+#
+, withWayland ? true, wayland, wayland-protocols
+
+#
+, withEspeak ? true, espeak-classic
+
+#
+, withLibmagic ? true, file
+
+#
+, withTesseractAndLeptonica ? true, tesseract, leptonica
+
+#
+, withVNC ? true, libvncserver
+
+# Output debug information
+, debugging ? false }:
+
+let
+  name = "arcan";
+
+  srcs = {
+    arcan = fetchFromGitHub {
+      owner = "letoram";
+      repo = name;
+      rev = "613d613d966b7224c41ae49061e5dcfaf19b3bea";
+      sha256 = "sha256-teqKrNdLeO/3x6Na8EgJCS0SK9K055czP2pFPmNUuzs=";
+    };
+
+    openal = fetchFromGitHub {
+      owner = "letoram";
+      repo = "openal";
+      rev = "1c7302c580964fee9ee9e1d89ff56d24f934bdef"; # master
+      sha256 = "0dcxcnqjkyyqdr2yk84mprvkncy5g172kfs6vc4zrkklsbkr8yi2";
+    };
+  };
+in stdenv.mkDerivation rec {
+  pname = name;
+
+  version = "0.6.1pre1+unstable=2021-05-27";
+
+  src = srcs.arcan;
+
+  postUnpack = ''
+    # ln -sv ${luajit.src} $sourceRoot/external/git/luajit
+    ln -sv ${freetype.src} $sourceRoot/external/git/freetype
+    ln -sv ${srcs.openal} $sourceRoot/external/git/openal
+    ln -sv ${libuvc.src} $sourceRoot/external/git/libuvc
+
+    # Generate manpages
+    pushd $sourceRoot/doc/
+    mkdir mantmp
+    ruby docgen.rb mangen
+    popd
+  '';
+
+  postPatch = ''
+    substituteInPlace ./src/platform/posix/paths.c \
+      --replace "/usr/bin" "$out/bin" \
+      --replace "/usr/share" "$out/share"
+
+    substituteInPlace ./src/CMakeLists.txt \
+      --replace "SETUID" ""
+  '';
+
+  nativeBuildInputs = [ cmake pkg-config ruby ];
+
+  buildInputs = [
+    openal
+    freetype
+    sqlite
+    libuvc
+    libGL
+    SDL2
+    libdrm
+    libxkbcommon
+    mesa
+    valgrind
+    luajit
+    libusb1
+    libvlc
+    xz
+    libffi
+    libX11
+    libxcb
+    libXcomposite
+    libXfixes
+    xcbutil
+    xcbutilwm
+    libXau
+    libXdmcp
+    harfbuzzFull
+  ] ++ lib.optionals withFFmpeg [ ffmpeg-full ]
+    ++ lib.optionals withWayland [ wayland wayland-protocols ]
+    ++ lib.optionals withEspeak [ espeak-classic ]
+    ++ lib.optionals withLibmagic [ file ]
+    ++ lib.optionals withTesseractAndLeptonica [ tesseract leptonica ]
+    ++ lib.optionals withVNC [ libvncserver ];
```

Comment:
Why did you merged the lines here? The previous format was better, one attribute per line.

Reply:
I thought it looked neat, but I've split them with one attribute for each line.